EIPS/eip-7002.md

```diff
@@ -147,137 +138,402 @@ def fake_exponential(factor: int, numerator: int, denominator: int) -> int:
+        sstore(VALIDATOR_EXIT_ADDRESS, EXIT_MESSAGE_QUEUE_HEAD_STORAGE_SLOT, new_queue_head_index)
+
+    return exits
 
 def update_excess_exits():
-    previous_excess_exits = sload(VALIDATOR_EXIT_PRECOMPILE_ADDRESS, EXCESS_EXITS_STORAGE_SLOT)
-    exit_count = sload(VALIDATOR_EXIT_PRECOMPILE_ADDRESS, EXIT_COUNT_STORAGE_SLOT)
+    previous_excess_exits = sload(VALIDATOR_EXIT_ADDRESS, EXCESS_EXITS_STORAGE_SLOT)
+    exit_count = sload(VALIDATOR_EXIT_ADDRESS, EXIT_COUNT_STORAGE_SLOT)
 
     new_excess_exits = 0
     if previous_excess_exits + exit_count > TARGET_EXITS_PER_BLOCK:
         new_excess_exits = previous_excess_exits + exit_count - TARGET_EXITS_PER_BLOCK
     
-    sstore(VALIDATOR_EXIT_PRECOMPILE_ADDRESS, EXCESS_EXITS_STORAGE_SLOT, new_excess_exits)
+    sstore(VALIDATOR_EXIT_ADDRESS, EXCESS_EXITS_STORAGE_SLOT, new_excess_exits)
     
 def reset_exit_count():
-    sstore(VALIDATOR_EXIT_PRECOMPILE_ADDRESS, EXIT_COUNT_STORAGE_SLOT, 0)
+    sstore(VALIDATOR_EXIT_ADDRESS, EXIT_COUNT_STORAGE_SLOT, 0)
 ```
 
+#### Bytecode
+
+```
+caller
+push20 0xfffffffffffffffffffffffffffffffffffffffe
+eq
+push1 0x90
+jumpi
+
+calldatasize
+iszero
+iszero
+push1 0x28
+jumpi
+
+push0
+sload
+push0
+mstore
+push1 0x20
+push0
+return
+
+jumpdest
+calldatasize
+push1 0x30
+eq
+iszero
+push2 0x0152
+jumpi
+
+push1 0x11
+push0
+sload
+push1 0x01
+dup3
+mul
+push1 0x01
+swap1
+push0
+
+jumpdest
+push0
+dup3
+gt
+iszero
+push1 0x59
+jumpi
+
+dup2
+add
+swap1
+dup4
+mul
+dup5
+dup4
+mul
+swap1
+div
+swap2
+push1 0x01
+add
+swap2
+swap1
+push1 0x3e
+jump
+
+jumpdest
+swap1
+swap4
+swap1
+div
+callvalue
+lt
+push2 0x0152
+jumpi
+
+push1 0x01
+sload
+push1 0x01
+add
+push1 0x01
+sstore
+push1 0x03
+sload
+dup1
+push1 0x03
+mul
+push1 0x04
+add
+caller
+dup2
+sstore
+push1 0x01 
+add
+push0
+calldataloa
+dup2
+sstore
+push1 0x01 
+add
+push1 0x20
+calldataloa
+swap1
+sstore
+push1 0x01 
+add
+push1 0x03
+sstore
+stop
+
+jumpdest
+push1 0x03
+sload
+push1 0x02 
+sload
+dup1
+dup3
+sub
+dup1
+push1 0x10
+gt
+push1 0xa4
+jumpi
+
+pop
+push1 0x10
+
+jumpdest
+push0
+
+jumpdest
+dup2
+dup2
+eq
+push2 0x010c
+jumpi
+
+dup3
+push1 0x03
+mul
+dup2
+push1 0x03
+mul
+push1 0x04
+add
+add
+dup1
+sload
+swap1
+push1 0x01 
+add
+dup1
+sload
+swap1
+push1 0x01 
+add
+sload
+swap2
+push1 0x60
+shl
+dup2
+push1 0xa0 
+shr
+or
+dup4
+push1 0x44
+mul
+mstore
+push1 0x60
+shl
+dup2
+push1 0xa0 
+shr
+or
+dup3
+push1 0x44
+mul
+push1 0x20
+add
+mstore
+push20 0xff
+and
+push1 0x60
+shl
+dup2
+push1 0x44
+mul
+push1 0x40
+add
+mstore
+push1 0x01 
+add
+push1 0xa6
+jump
+
+jumpdest
+swap2
+add
+dup1
+swap3
+eq
+push2 0x011
+jumpi
+
+swap1
+push1 0x02 
+sstore
+push2 0x012
+jump
+
+jumpdest
+swap1
+pop
+push0
+push1 0x02 
+sstore
+push0
+push1 0x03
+sstore
+
+jumpdest
+push0
+sload
+push1 0x01 
+sload
+push1 0x02 
+dup3
+dup3
+add
+gt
+push2 0x014
+jumpi
+
+pop
+pop
+push0
+push2 0x014
```

Comment:
This offset seems wrong. If we assume the offset in line#454 (`0x0140`) is correct, then this one should be `0x0140 + 0x6`